File: producer.py
# producer.py — Places a single order into SQS
# Run: python producer.py
# Install: pip install boto3

import boto3
import json
import random
import os
from datetime import datetime, timezone

sqs = boto3.client("sqs", region_name="us-east-1")

QUEUE_URL = os.environ["SQS_QUEUE_URL"]

PRODUCTS = [
    "Wireless Headphones",
    "Mechanical Keyboard",
    "USB-C Hub",
    "Webcam 4K",
    "Laptop Stand",
]


def generate_order():
    return {
        "orderId": f"ORD-{random.randint(100000, 999999)}",
        "product": random.choice(PRODUCTS),
        "quantity": random.randint(1, 3),
        "userId": f"USR-{random.randint(1000, 9999)}",
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }


def place_order():
    order = generate_order()

    response = sqs.send_message(
        QueueUrl=QUEUE_URL,
        MessageBody=json.dumps(order),
        MessageAttributes={
            "Source": {
                "DataType": "String",
                "StringValue": "shopq-frontend",
            }
        },
    )

    print("✅ Order placed successfully!")
    print(f"   Order ID  : {order['orderId']}")
    print(f"   Product   : {order['product']}")
    print(f"   Quantity  : {order['quantity']}")
    print(f"   MessageId : {response['MessageId']}")
    print(f"   Queued at : {order['timestamp']}")


if __name__ == "__main__":
    place_order()

File: producer.js
// producer.js — Places a single order into SQS
// Run: node producer.js

const { SQSClient, SendMessageCommand } = require("@aws-sdk/client-sqs");

const client = new SQSClient({ region: "ap-south-1" });

const QUEUE_URL = process.env.SQS_QUEUE_URL; // export SQS_QUEUE_URL=<your-queue-url>

const PRODUCTS = [
  "Wireless Headphones",
  "Mechanical Keyboard",
  "USB-C Hub",
  "Webcam 4K",
  "Laptop Stand",
];

function generateOrder() {
  return {
    orderId: "ORD-" + Math.floor(Math.random() * 900000 + 100000),
    product: PRODUCTS[Math.floor(Math.random() * PRODUCTS.length)],
    quantity: Math.floor(Math.random() * 3 + 1),
    userId: "USR-" + Math.floor(Math.random() * 9000 + 1000),
    timestamp: new Date().toISOString(),
  };
}

async function placeOrder() {
  const order = generateOrder();

  const command = new SendMessageCommand({
    QueueUrl: QUEUE_URL,
    MessageBody: JSON.stringify(order),
    MessageAttributes: {
      Source: {
        DataType: "String",
        StringValue: "shopq-frontend",
      },
    },
  });

  const response = await client.send(command);

  console.log("✅ Order placed successfully!");
  console.log("   Order ID  :", order.orderId);
  console.log("   Product   :", order.product);
  console.log("   Quantity  :", order.quantity);
  console.log("   MessageId :", response.MessageId);
  console.log("   Queued at :", order.timestamp);
}

placeOrder().catch(console.error);